src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -456,6 +462,110 @@ class spell_sha_earthen_rage_proc_aura : public AuraScript
+// 61882 - Earthquake
+class spell_sha_earthquake : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_AFTERSHOCK, SPELL_SHAMAN_AFTERSHOCK_ENERGIZE });
+    }
+
+    void HandleAfterCast()
+    {
+        int32 energize = *GetSpell()->GetPowerTypeCostAmount(POWER_MAELSTROM);
+
+        if (AuraEffect* aftershock = GetCaster()->GetAuraEffect(SPELL_SHAMAN_AFTERSHOCK, EFFECT_0))
+            if (roll_chance_f(aftershock->GetAmount()))
+                GetCaster()->CastSpell(GetCaster(), SPELL_SHAMAN_AFTERSHOCK_ENERGIZE, CastSpellExtraArgs(energize)
+                    .AddSpellBP0(energize));
+    }
+
+    void Register() override
+    {
+        AfterCast += SpellCastFn(spell_sha_earthquake::HandleAfterCast);
+    }
+};
+
+// 61882 - Earthquake aura
+class spell_sha_earthquake_aura : public AuraScript
+{
+    PrepareAuraScript(spell_sha_earthquake_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_EARTHQUAKE, SPELL_SHAMAN_EARTHQUAKE_TICK });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/)
+    {
+        std::vector<AreaTrigger*> areaTriggers = GetCaster()->GetAreaTriggers(SPELL_SHAMAN_EARTHQUAKE);
+
+        for (AreaTrigger* at : areaTriggers)
+            GetTarget()->CastSpell(at->GetPosition(), SPELL_SHAMAN_EARTHQUAKE_TICK, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_sha_earthquake_aura::HandlePeriodic, EFFECT_1, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 77478 - Earthquake tick
+class spell_sha_earthquake_tick : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake_tick);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_SHAMAN_ELEMENTAL_SHAMAN,
+            SPELL_SHAMAN_BOTTLE_OF_SWIRLING_MAELSTROM,
+            SPELL_SHAMAN_EARTHQUAKE_KNOCKING_DOWN
+        });
+    }
+
+    void CalcDamage(SpellEffIndex /*effIndex*/)
+    {
+        const Player* player = GetCaster()->ToPlayer();
+        float versaDmgMod = 1.0f;
+        // $mawPower = ${ 1 + ($329580m5 / 100) }
+        // $damage = ${ $SPN * 0.391 * $d / $t2 * (1 + $@versadmg) * (($137040s1 + 100) / 100) * $<mawPower> }
+        AddPct(versaDmgMod, player->GetRatingBonusValue(CR_VERSATILITY_DAMAGE_DONE) + float(player->GetTotalAuraModifier(SPELL_AURA_MOD_VERSATILITY)));
+
+        uint32 dmg = player->m_activePlayerData->ModDamageDonePos[SPELL_SCHOOL_NATURE] * 0.391f * versaDmgMod;
+
+        if (AuraEffect* aurEff = player->GetAuraEffect(SPELL_SHAMAN_ELEMENTAL_SHAMAN, EFFECT_0))
+        {
+            float dmgMod = 1.0f;
+            dmg *= AddPct(dmgMod, aurEff->GetAmount());
+        }
+
+        if (AuraEffect* aurEff = player->GetAuraEffect(SPELL_SHAMAN_BOTTLE_OF_SWIRLING_MAELSTROM, EFFECT_4))
```

Comment:
Are we sure we want to start adding Torghast Anima Powers values in here? Should it not be wiser to make the effect of such aura complement the ticks instead rather than raw calculating on its tick spell?

Reply:
Yeah, you may be right. I gonna try another solution. Thanks for commenting this.